R/constructor.R

```diff
@@ -63,6 +64,12 @@ constructor_args <- function(parent, properties = list()) {
     function(name) prop_default(properties[[name]]))
   )
 
+  is_dots <- vlapply(self_args, identical, quote(...))
+  if (any(is_dots)) {
```

Comment:
> the first argument to every constructor should be `...`

I think this is going to be too annoying in practice. Many classes will only have a small handful of properties, and being unable to positionally match 1, 2, or 3 would be cumbersome. 

> Otherwise you end up with a mix of arguments that can/can't be supplied by position.

I think #446 might solve this by allowing the user to specify a property that is unset by default but still present as a named arg in the constructor formals that can match positionally.

Reply:
In that case, I think you should make `...` the last argument, if it's present.

Reply:
I updated the code so `...` is added at the end of the constructor. Note, deprecating this way now will potentially change the position of later arguments  E.g., 

```r
# before deprecating: 
function(firstName, lastName, country)

# after deprecating `firstName` and `lastName`:
function(name, country, ...)
# whereas with the previous approach, this was possible:
function(name, ..., country)
```

Reply:
Maybe we should chat a bit more about this live, because I don't quite see the problem.